agent/agent.go

```diff
@@ -164,8 +168,35 @@ func (a *Agent) generateNomadClient() error {
 // reload triggers the reload of sub-routines based on the operator sending a
 // SIGHUP signal to the agent.
 func (a *Agent) reload() {
+	a.logger.Info("reloading Autoscaler configuration")
+
+	// Reload config files from disk.
+	newCfg, err := config.LoadPaths(a.configPaths)
+	if err != nil {
+		a.logger.Error("failed to reload Autoscaler configuration", "error", err)
+		return
+	}
+
+	a.config = newCfg
+	a.nomadCfg = nomadHelper.MergeDefaultWithAgentConfig(newCfg.Nomad)
+
+	if err := a.generateNomadClient(); err != nil {
+		a.logger.Error("failed to reload Autoscaler configuration", "error", err)
```

Comment:
i wasn't sure about this error message being the same as above, but it look `generateNomadClient` adds "failed to instantiate Nomad client" to any message from Nomad, so that's fine